README.md

```diff
@@ -245,10 +245,104 @@ If you experienced a compatibility issue with any of the following plugins, plea
 - [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin)
 - [HtmlWebpackPlugin](https://github.com/jantimon/html-webpack-plugin)
 
+### options.contentScript.experimental_output
+
+**This is an experimental API. API might change at any time. Please provide feedback!**
+
+#### TLDR: How to use this
+
+##### string
+
+If you don't strictly rely on [run_at](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts#run_at),
+set it as the following
+
+```js
+export default {
+  entry: {
+    myContentScript: 'src/contentScript.ts',
+  },
+  // ...
+  plugins: [
+    // ...
+    new WebExtensionPlugin({
+      // ...
+      contentScript: {
+        experimental_output: {
+          myContentScript: 'cs.js'
+        },
+      },
+    })
+  ]
+}
+```
+
+```jsonc
+{
+  // ...
+  "content_scripts": [
+    {
+      "matches": ["..."],
+      "js": ["cs.js"]
+    }
+  ]
+}
+```
+
+##### function
+
+If you cannot use asynchronous loading, setup like below.
+This setup requires you having a `manifest.json` being emitted.
+
+```js
+export default {
+  entry: {
+    myContentScript: 'src/contentScript.ts',
+  },
+  // ...
+  plugins: [
+    // ...
+    new WebExtensionPlugin({
+      // ...
+      contentScript: {
+        experimental_output: {
+          myContentScript: (manifest, list) => {
+            manifest.content_scripts[0].js = list
```

Comment:
could this handle more than one content_script defined in the manifest file? like which list would this be if I have a `manifest.content_scripts[1].js`?

Reply:
Yes, this is a function, so you can write any code to change your `manifest.json` (if you defined it via CopyPlugin or other webpack plugins).

```js
{
myContentScript: (manifest, list) => {
    manifest.content_scripts[0].js = list
    manifest.content_scripts[1].js = list
},

myContentScript2: (manifest, list) => {
    manifest.content_scripts[3].js = list
},
}
```